### PR TITLE
config: update `.editorconfig` parameters according to current format of files

### PR DESCRIPTION
**Pull request by davorpa**
## What does this PR do?
Improve repo

### Description

Update `.editorconfig` parameters according to values provided by @SethFalco in #5564 (commit e9f7dff3be87e640a72f97bd4eec1be79ec2b43a) and extend it.

- 2 spaces indent for `yaml` or `yml` files
- 2 spaces indent for `json` & `jsonc` (commented) files
- avoid strip trailing spaces in [any kind of markdown files](https://superuser.com/questions/249436/file-extension-for-markdown-files)

Moreabout editorconfig syntax at: https://editorconfig.org/

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md)
- [x] Search for duplicates.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Review by LuigiImVector — COMMENTED** [review]
Why did you use `;` for the rule comments and `#` for the file description? And why did you include the python, bat, powershell files if they are not used?

**Comment by davorpa** [comment]
> Why did you use `;` for the rule comments and `#` for the file description?

According documentation comments can be addressed by `#` or `;` without side effects: https://editorconfig-specification.readthedocs.io/#file-format

> And why did you include the python, bat, powershell files if they are not used?

Sorry, I have some scripts in my local environment testing things with nodejs and remark. Let's decide if revert fd9a837b6568f213f2df26cc293f72257dd627c4, 5516f58a0a4c4fd720a05574a09cfe56c9f8f83f, 20bac81b2d34bb770dac2a1959bac3b3d3202326, 4b2c37bc188e834709b79925a67bd2380aa99649 commits is needed.

Its parametrization is taken from known examples provided in: https://github.com/editorconfig/editorconfig/wiki/Projects-Using-EditorConfig

Moreover... I update the link pointing to this repo in that wiki :stuck_out_tongue_winking_eye:

**Comment by LuigiImVector** [comment]
> Sorry, I have some scripts in my local environment testing things with nodejs and remark. Let's decide if revert [fd9a837](https://github.com/EbookFoundation/free-programming-books/commit/fd9a837b6568f213f2df26cc293f72257dd627c4), [5516f58](https://github.com/EbookFoundation/free-programming-books/commit/5516f58a0a4c4fd720a05574a09cfe56c9f8f83f), [20bac81](https://github.com/EbookFoundation/free-programming-books/commit/20bac81b2d34bb770dac2a1959bac3b3d3202326), [4b2c37b](https://github.com/EbookFoundation/free-programming-books/commit/4b2c37bc188e834709b79925a67bd2380aa99649) commits is needed.

I think they need to be deleted to have a cleaner code, maybe wait for a review from someone more experienced

**Comment by SethFalco** [comment]
I just use the same EditorConfig file in all my projects, regardless of the contents of the repository.
I don't see the harm in including the extra, especially since when or if a batch file were to be added, said person probably wouldn't update the EditorConfig after. On the other hand, it does seem a bit cluttered tbh. 

The above is an opinion, and nothing more.

---

However, I think most if not all the comments should be entirely removed. (EditorConfig is very self-explanatory as it is.)

```
; JSON files - compress a little more
[*.{json,jsonc}]
indent_size = 2
```

And the comments aren't necessarily accurate either. "Compress" compared to what?
Coding conventions shouldn't be commented with an action, since that action may not apply to how everyone writes JSON.

In other words, the EditorConfig _describes_ how the code should look to editors.
In this case, if it compresses or inflates the JSON file depends entirely on if the developer normally does 1 space, or 4 spaces.

**Comment by SethFalco** [comment]
I do also wonder where all the file extensions came from, for example Markdown.

```
; Markdown files - preserve whitespaces
[*.{md,mkd,mkdn,mdwn,mdown,markdown}]
trim_trailing_whitespace = false
```

On Wikipedia:
> `.md`, `.markdown`
> 
> — https://en.wikipedia.org/wiki/Markdown

This is also another case where the comment seems a bit redundant. ^-^'
It only has 1 rule, which is to not trim trailing whitespace. We can tell it's for preserving the whitespace.

**Comment by davorpa** [comment]
Thanks for the starter template. Adopted a bit.

Comments are in part for my future self (I have a fish memory sometimes)

Better now after 520d0e9 or completely remove not used?

**Comment by LuigiImVector** [comment]
> Better now after [520d0e9](https://github.com/EbookFoundation/free-programming-books/commit/520d0e91e28b748b414686b126ea079db3ba9d82) or completely remove not used?

Comments are better now even if i think you can remove the unused ones

**Comment by davorpa** [comment]
What more I need to go ahead with the PR?

**Comment by LuigiImVector** [comment]
> What more I need to go ahead with the PR?

I think the only problems are the "excess" of information it contains: comments and (not yet) used extensions.

Comments can be left as they are in my opinion, for the extensions (ps1, js, ts...) I find that they are currently useless but could be used in the future, if no one opposes pr can be merged.

**Comment by davorpa** [comment]
I clean it a bit more.

**Comment by LuigiImVector** [comment]
LGTM
